src/main/java/org/apache/commons/io/input/XmlStreamReader.java

```diff
@@ -746,6 +745,7 @@ String calculateHttpEncoding(@UnderInitialization(java.io.Reader.class) XmlStrea
                 }
                 final Matcher m = ENCODING_PATTERN.matcher(prolog);
                 if (m.find()) {
+                    assert m.group(1) != null : "@AssumeAssertion: ENCODING_PATTERN has capturing group 1";
```

Comment:
Hi Mike,

Checker continue to issue following error, even after asserting non-null value for m.group(1)
```
XmlStreamReader.java:[749,38] error: [dereference.of.nullable] dereference of possibly-null reference m.group(1)
``` 
In annotated jdk `Matcher.group(int)` is annotated as `@SideEffectFree`, For same matcher and arguments `group()` returns identical pattern, which supports deterministic behaviour of this method. Should annotation for `Matcher.group(int)` changed to `@Pure`.

Reply:
You are right.  Please fix the annotation, perhaps by introducing a `@NonNull` local variable.

`group()` is pure up to `equals()` but not `==`, so it cannot be annotated as `@Pure`, unfortunately.

Reply:
This warning needs to be suppressed as m.group(1) is directly used in expression.